v2/internal/controllers/crd_servicebus_namespace_basic_test.go

```diff
@@ -53,10 +53,7 @@ func Test_ServiceBus_Namespace_Basic_CRUD(t *testing.T) {
 	tc.DeleteResourceAndWait(namespace)
 
 	// Ensure that the resource was really deleted in Azure
-	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(servicebus.APIVersion_Value))
-	tc.Expect(err).ToNot(HaveOccurred())
-	tc.Expect(retryAfter).To(BeZero())
-	tc.Expect(exists).To(BeFalse())
+	tc.ExpectResourceIsDeletedInAzure(armId, string(servicebus.APIVersion_Value))
```

Comment:
It used to be that we could reliably say that when the ASO resource deletion completed, the resource in Azure was 100% gone. It seems like for most services that is still the case, but some services you can still GET the resource for a very short time even after the async operation ID has reported that the delete succeeded. Possibly this is due to some caching layer for those services and possibly it's just their API behaving a bit weird.

While we _could_ continue to maintain that invariant by explicitly doing a GET and waiting for the resource to have disappeared in the operator before removing it, I didn't think it made sense to do so since it costs it's an extra request that we otherwise wouldn't need.

AFAIK, all of the SDKs do what we're doing here (poll the operation until its done). They don't check that GET 404's after that, so we don't either.

That means that in the tests, we needed to change some "assert this is true _now_" to "assert this is true eventually". This is one such example.